mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -71,6 +71,26 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
 
   /** @group getParam */
   def getItemCol: String = $(itemCol)
+
+  /**
+   * Param for strategy for dealing with unknown or new users/items at prediction time.
+   * This may be useful in cross-validation or production scenarios, for handling user/item ids
+   * the model has not seen in the training data.
+   * Supported values:
+   * - "nan": predicted value for unknown ids will be NaN.
```

Comment:
Nope - easier to type? :) also seems consistent with most other string
params which seem to be lower case (eg "rmse", "map" etc).
On Wed, 11 May 2016 at 19:24, Holden Karau <notifications@github.com> wrote:

> In mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala
> <https://github.com/apache/spark/pull/12896#discussion_r62889428>:
>
> > @@ -71,6 +71,26 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
> >
> >    /** @group getParam */
> >    def getItemCol: String = $(itemCol)
> > +
> > +  /**
> > +   * Param for strategy for dealing with unknown or new users/items at prediction time.
> > +   * This may be useful in cross-validation or production scenarios, for handling user/item ids
> > +   * the model has not seen in the training data.
> > +   * Supported values:
> > +   * - "nan": predicted value for unknown ids will be NaN.
>
> Any reason why "nan" rather than "NaN"?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12896/files/b71dcaca39d5f10b484932eaf25e290c7796d93b#r62889428>
>